docs/quickstarts/google.rst

```diff
@@ -95,3 +95,54 @@ you can use all the normal ``requests`` methods, like
 :meth:`~requests.Session.get` and :meth:`~requests.Session.post`,
 to make HTTP requests. If you only specify the path component of the URL,
 the domain will default to ``https://www.googleapis.com``.
+
+Online vs offline applications
+------------------------------
+
+Google distinguishes between online and offline applications. Online applications
+are applications that only act when the user is at their keyboard, say a web
+application that only calls Google API's as the user is interacting with your
+application. Offline applications are applications that can act on behalf of the
+user while they're not at their keyboard, for example something that runs backups
+on behalf of the user at a specific point in time.
+
+The blueprint is configured by default to get an online token. This means you'll get
+a token that's valid for about an hour and after that you'll need to get a new token,
+by putting the user through the OAuth flow again. If instead you request an offline
+token you'll also be given a refresh token that your application can use to request
+a new, short lived, token without the user needing to do anything.
+
+.. warning::
+
+   The refresh token is only returned the first time you put a user through the OAuth
+   flow. This means you need to store the refresh token in a persistent fashion in
+   order to ensure your application can renew the access token. It's therefore not
+   advisable to store this using Flask's regular session, which uses a cookie. Though
+   the session can be marked as persistent that won't save you if the user decides
+   to clear their browser cache for example.
+
+It's tempting to just request an offline token but if your application doesn't act
+on behalf of the user while they're not at their keyboard this is in bad form. Instead,
```

Comment:
"Bad form" is not a good reason to avoid offline tokens. Instead, it's better to talk about security. A core principal of security is to only provide the minimal set of permissions necessary to get the job done. If your application doesn't need offline access, then it shouldn't request it, because offline tokens are a greater target for attackers. If you only use online tokens, that makes it much harder for an attacker to get access to the user's account.